src/addrman_impl.h

```diff
@@ -32,6 +32,13 @@ static constexpr int ADDRMAN_NEW_BUCKET_COUNT{1 << ADDRMAN_NEW_BUCKET_COUNT_LOG2
 static constexpr int32_t ADDRMAN_BUCKET_SIZE_LOG2{6};
 static constexpr int ADDRMAN_BUCKET_SIZE{1 << ADDRMAN_BUCKET_SIZE_LOG2};
 
+/**
+ * User-defined type for the internally used nIds
+ * This used to be int, making it feasible for attackers to cause an overflow,
+ * see https://bitcoincore.org/en/2024/07/31/disclose-addrman-int-overflow/
+ */
+using nid_type = int64_t;
```

Comment:
[Good](https://www.aristeia.com/Papers/C++ReportColumns/sep95.pdf) to see that a signed type is used, rather than an unsigned one in the initial version of this PR.

Reply:
interesting pdf, thanks! In this particular case there was no choice because `nId`s in `vvNew` and `vvTried` are [initialised](https://github.com/bitcoin/bitcoin/blob/2aff9a36c352640a263e8b5de469710f7e80eb54/src/addrman.cpp#L109-L125) to `-1`.